crates/ty_python_semantic/resources/mdtest/generics/legacy/classes.md

```diff
@@ -732,6 +732,14 @@ class Base(Generic[T]): ...
 class Sub(Base["Sub"]): ...
 
 reveal_type(Sub)  # revealed: <class 'Sub'>
+
+U = TypeVar("U")
+
+class Base2(Generic[T, U]): ...
+
+# TODO: no error
+# error: [unsupported-base] "Unsupported class base with type `<class 'Base2[Sub2, U@Sub2]'> | <class 'Base2[Sub2[Unknown], U@Sub2]'>`"
```

Comment:
I see this occurring in ecosystem projects, so we will want to follow up on this soon; users may experience it as a regression.

> cycle normalization takes a union with the type from the previous cycle

It seems like maybe we could address this with a targeted fix in the cycle recovery for `explicit_bases` query? For class literal types we could skip unioning, or union the type parameters instead if the outer type is the same and the outer type is covariant in its type parameters? Not sure on details, would need to experiment.